Sources/StreamChatUI/ChatChannelList/ChatChannelListVC_Tests.swift

```diff
@@ -46,50 +86,104 @@ class ChatChannelListVC_Tests: XCTestCase {
+    }
+    
+    func test_defaultAppearance() {
+        mockedChannelListController.simulate(
+            channels: mockedChannels,
+            changes: []
         )
-        let channelWithLongTextAndUnreadCount = ChatChannel.mock(
-            cid: .init(type: .messaging, id: "test_channel3"),
-            name: "This is a channel with a big name. Really big.",
-            imageURL: TestImages.yoda.url,
-            lastMessageAt: .init(timeIntervalSince1970: 1_611_951_528_000),
-            unreadCount: .mock(messages: 4),
-            latestMessages: [
-                ChatMessage.mock(
-                    id: "1", text: "This is a long message. How the UI will adjust?", author: .mock(id: "Vader2")
-                )
-            ]
+        AssertSnapshot(vc, isEmbeddedInNavigationController: true)
+    }
+
+    func test_appearanceCustomization_usingUIConfig() {
+        class TestView: CellSeparatorReusableView {
+            override func setUpAppearance() {
+                super.setUpAppearance()
+
+                separatorView.backgroundColor = UIColor.gray
+            }
+
+            override func setUpLayout() {
+                super.setUpLayout()
+
+                separatorView.leadingAnchor.constraint(equalTo: leadingAnchor, constant: 15).isActive = true
+            }
+        }
+
+        var config = UIConfig()
+        config.channelList.cellSeparatorReusableView = TestView.self
+
+        vc.uiConfig = config
+
+        mockedChannelListController.simulate(
+            channels: mockedChannels,
+            changes: []
         )
-        let channelWithMultipleMessages = ChatChannel.mock(
-            cid: .init(type: .messaging, id: "test_channel4"),
-            name: "Channel 4",
-            imageURL: TestImages.vader.url,
-            lastMessageAt: .init(timeIntervalSince1970: 1_611_951_529_000),
-            latestMessages: [
-                ChatMessage.mock(id: "2", text: "Hello", author: .mock(id: "Vader")),
-                ChatMessage.mock(id: "1", text: "Hello2", author: .mock(id: "Vader2"))
-            ]
+        AssertSnapshot(vc, isEmbeddedInNavigationController: true, variants: .onlyUserInterfaceStyles)
+    }
+
+    func test_appearanceCustomization_usingAppearanceHook() {
+        class TestSeparatorView: CellSeparatorReusableView {
+            override func setUpAppearance() {
+                super.setUpAppearance()
+                separatorView.backgroundColor = UIColor.orange
+            }
+        }
+
+        class TestView: TestChatChannelListVC {}
+        TestView.defaultAppearance {
+            $0.createNewChannelButton.tintColor = UIColor.orange
+            if let listLayout = $0.collectionViewLayout as? ListCollectionViewLayout {
+                listLayout.separatorHeight = 4
+                listLayout.register(TestSeparatorView.self, forDecorationViewOfKind: listLayout.separatorIdentifier)
+            }
+        }
```

Comment:
Not sure how I feel about this, it is not very friendly right now to change both the color and separator height, there are two steps needed, which I guess is not expected, I think we should be able to customize both the color and height in the same place. WDYT? 🤔

Reply:
Btw, this is true only for subclassing. When using UIConfig, it is easier.

Reply:
I see what you mean @nuno-vieira. You could probably do this if you really needed to have this in one place:
```swift
        TestView.defaultAppearance {
             $0.createNewChannelButton.tintColor = UIColor.orange
             if let listLayout = $0.collectionViewLayout as? ListCollectionViewLayout {
                 listLayout.separatorHeight = 4
                 listLayout.register(TestSeparatorView.self, forDecorationViewOfKind: listLayout.separatorIdentifier)
             }
             TestSeparatorView.defaultAppearance { 
               $0.separatorView.backgroundColor = .orange
             }
         }
```
We don't have access to the separator instances directly AFAIK. The collection view manages them automatically.

If it's ugly/dangerous/confusing you need to decide for yourself 🤷‍♂️